cmd/saturn-l2/main.go

```diff
@@ -244,6 +257,23 @@ func main() {
 		log.Info("finished tearing down all l1 connections")
 	})
 
+	// if we fail to connect to any of the L1s after exhausting all retries; error out and exit.
+	go func() {
+		for i := 0; i < len(l1IPAddrs); i++ {
+			select {
+			case <-failedL1Ch:
+			case <-ctx.Done():
+				return
+			}
+		}
+		log.Error("failed to connect to any of the L1s after exhausting all attempts; shutting down")
+		fmt.Println("\n ERROR: Saturn node failed to connect to the network and has exhausted all retry attempts")
```

Comment:
@bajtos Please feel free to change this error message that makes sense for the user.

Reply:
The message LGTM 👍🏻 

What is the reason for the leading newline and an extra space character?

```suggestion
		fmt.Println("ERROR: Saturn node failed to connect to the network and has exhausted all retry attempts")
```